src/properties.c

```diff
@@ -447,17 +442,17 @@ static int set_schema(struct knot_thing *thing, int fd, char *group_id,
 	l_free(name);
 
 	rc = storage_read_key_int(fd, group_id, SCHEMA_VALUE_TYPE, &aux);
-	if (rc <= 0)
+	if (rc < 0)
 		return -EINVAL;
 	schema_aux.value_type = aux;
 
 	rc = storage_read_key_int(fd, group_id, SCHEMA_UNIT, &aux);
-	if (rc <= 0)
+	if (rc < 0)
 		return -EINVAL;
 	schema_aux.unit = aux;
 
 	rc = storage_read_key_int(fd, group_id, SCHEMA_TYPE_ID, &aux);
```

Comment:
I would recommend replacing the 'rc' and 'fd' variables with names with more details